lib/react_client.dart

```diff
@@ -496,7 +496,7 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
           ..jsThis = jsThis
           ..props = new JsBackedMap.backedBy(jsThis.props)
           ..context = _unjsifyNewContext(jsThis.context);
-
+        component.init();
```

Comment:
I believe there is more to it than this...

From the ticket AC:

```dart
component.init();

// Sync the JS component's state the component's state
if (jsThis.dartComponent.state != null) {
  jsThis.state = new JsBackedMap.from(jsThis.dartComponent.state).jsMap;
}
```

Which also indicates to me that lines 495-498 should probably come after `component.init()` et. al. like so:

```dart
component.init();

// Sync the JS component's state the component's state
if (jsThis.dartComponent.state != null) {
  jsThis.state = new JsBackedMap.from(jsThis.dartComponent.state).jsMap;
}

component
  ..jsThis = jsThis
  ..props = new JsBackedMap.backedBy(jsThis.props)
  ..context = _unjsifyNewContext(jsThis.context);

return component;
```

Does that look right to you @greglittlefield-wf @kealjones-wk ?